fastlane/Fastfile

```diff
@@ -658,8 +664,13 @@ REPOSITORY_NAME="WordPress-Android"
   #####################################################################################
   desc "Downloads translated metadata from GlotPress"
   lane :download_metadata_strings do |options|
-    app = options[:app] || 'wordpress'
-    paths = APP_SPECIFIC_VALUES[app.to_sym]
+    download_wordpress_metadata_strings(options)
+    download_jetpack_metadata_strings(options)
```

Comment:
This refactor is a little interesting. I can see that a lot of the logic is still shared between WordPress and Jetpack apps within this lane. So, I am curious why you opted to completely separate them.

Reply:
I am actually going to hold off on reviewing the relevant piece of the code, because I _think_ it might be worth to refactor and combine at least some bits. The lane itself is already long, so maybe it can be cut into some pieces that takes `app` as the parameter?

Reply:
I've started to split them because I've actually started to see too many things being different.
- The list of files (to which Annmarie will soon add the GlotPress keys for the copies used for promo screenshots, keys which are different between WP and JP too)
- But also the fact that WP uses the hacky fake en-GB=>en-US locale while JP can't, and that we need to do additional file copies instead
- the parameters to gp_download_metadata differ too

Sure we could refactor to mutualize sometime stuff (which honestly was how I kept it at first) but ultimately when I did that I realized there was not that much in common between those exceptions for locales and the different list of files and params.

---

We could also try to regroup them together again and move the params into the `APP_SPECIFIC_VALUES` constant that I've started to introduce (I plan to refactor some past existing constants to move them here too in a subsequent PR), but we'd still need the exception case for the hacky source locale… so I could give it a try but I'm not so sure that it's worth it? 🤷  (at least yet until further PRs refine and clean up the code once everything works)

Reply:
I see - that all sounds reasonable to me. What I was hoping we'd do is have small methods that with parameters when it made sense. I thought this would give us:

1. Reusable code (DRY)
2. Better code organization

When/if the extracted small method doesn't make sense to make it reusable, I thought we'd have 2 separate methods, one for `wordpress` and one for `jetpack`, so the logic separation is clear.

---

Since this PR is introducing the separation (and some code duplication) I thought it'd make sense to at least discuss it here. However, that doesn't mean we have to do it as part of this PR. If you feel comfortable with this setup right now and have plans to improve it, I'd be totally fine with that. If that's the case, I think it'd be good to open an issue for it just so we don't forget about it. What do you think?

Reply:
Yup makes sense and thanks for raising this, definitively worth discussing 👍 

I'm not sure I'll have time to tackle this and address your good feedback on this PR before going AFK, so just dropping a note here to let you know I saw your comment and plan to look at it later.

🤔 💭  PS: Since you offered to cover for me to do the 17.8 PlayStore submission during my AFK, I guess you'll be the one who might need to use this PR during submission (in case the metadata changed since last submission). If this PR doesn't make it in time before the submission, don't forget to at least double-check (and likely adjust manually) the Jetpack release notes XML, especially since, if this PR isn't applied by then, they will likely not contain the right list of locales for the JP release notes XML, and might also miss the `en-US` entry (which both are things this PR should fix)

Reply:
If you want, I can test the PR and merge it in today and you can follow up on it when you get back. What do you think?

Otherwise, it might be worth adding these items to the release checklist as otherwise we risk missing the extra steps. (for example, if I can't handle the release and Jeremy has to do it instead)

Reply:
Yeah it might be easier to merge this now and track the remaining refactoring points in issues that I'll tackle later (I already plan to refactor this once I've finished my other, higher priority work on all the Fastfiles accross projects, before coming back to Jetpack-Android-related work in the Fastfile next Sprint)

Not sure if it's worth updating the scenario with a note about this if we're gonna merge it by then (the note would only be useful for a single sprint until we get this PR validated)